In [1]:
import numpy as np
import pandas as pd
import re
 
#a = df['COST (£) charged to Wellcome (inc VAT when charged)'] = (df['COST (£) charged to Wellcome (inc VAT when charged)'].apply(lambda x: ''.join(list(filter(str.isdigit, str(x))))))


In [2]:
df = pd.read_csv('C:/Users/keith/Desktop/Thinkful/WELLCOME_APCspend2013_forThinkful.csv',encoding = "ISO-8859-1")
pd.set_option('display.max_columns', None) # disables trucated columns 
pd.set_option('display.max_rows', None) # disables trucated rows
df.head()
 

PMID/PMCID Publisher           Journal title  \
0                    NaN       CUP  Psychological Medicine   
1             PMC3679557       ACS       Biomacromolecules   
2  23043264  PMC3506128        ACS              J Med Chem   
3    23438330 PMC3646402       ACS              J Med Chem   
4   23438216 PMC3601604        ACS              J Org Chem   

                                       Article title  \
0  Reduced parahippocampal cortical thickness in ...   
1  Structural characterization of a Model Gram-ne...   
2  Fumaroylamino-4,5-epoxymorphinans and related ...   
3  Orvinols with mixed kappa/mu opioid receptor a...   
4  Regioselective opening of myo-inositol orthoes...   

  COST (£) charged to Wellcome (inc VAT when charged)  
0                                              £0.00   
1                                           £2381.04   
2                                            £642.56   
3                                            £669.64   
4                                            £685.88

In [3]:
journals = df['Journal title']

df['Journal title'] = journals.str.replace(' ', '')
df['Journal title'] = journals.str.replace('.', '')
df['Journal title'] = journals.apply(lambda x : re.sub('([Pp][Ll][Oo][Ss][Oo][Nn][Ee]|([Pp][Ll][Oo][Ss][1]))',"Plos One", str(x)))
df['Journal title'] = journals.apply(lambda x : re.sub('(\w*Biological\w*)',"Journal of Biological Chemistry", str(x)))
df['Journal title'] = journals.apply(lambda x : re.sub('(\w*[Nn]ucleic[Aa]c\w*)',"Nucleic Acids Research", str(x)))
df['Journal title'] = journals.apply(lambda x : re.sub('(\w*(nalAc\w*)|(PNAS)\w*)',"Proceedings of the National Academy of Sciences", str(x)))
df['Journal title'] = journals.apply(lambda x : re.sub('(\w*nMol\w*)',"Human Molecular Genetics", str(x))) 
df['Journal title'] = journals.apply(lambda x : re.sub('(\w*([Nn]euro[Iil]mage)\w*)',"Neuro Image", str(x))) 
 

In [7]:
journals.value_counts().head(5)

Plos One                                           207
Journal of Biological Chemistry                     64
Proceedings of the National Academy of Sciences     37
Neuro Image                                         31
Nucleic Acids Research                              26
Name: Journal title, dtype: int64

### The five most common journals.
Plos One count = 207<br>
Journal of Biological Chemistry count = 64<br>
Proceedings of the National Academy of Sciences count = 37<br>
Neuro Image = count 31<br>
Nucleic Acids Research count = 26<br>

In [5]:
cost = df['COST (£) charged to Wellcome (inc VAT when charged)']
cost = cost.str.replace('£', '')
cost = cost.str.replace('$','')
cost = cost.astype(float)
df['COST (£) charged to Wellcome (inc VAT when charged)'] = cost

In [6]:
a = df[df['Journal title']=='Plos One'] 
Plos_One = a['COST (£) charged to Wellcome (inc VAT when charged)']
print ('Plos One stats')
print (Plos_One.describe(),'\n')
b = df[df['Journal title']=='Journal of Biological Chemistry'] 
Journal_BC = b['COST (£) charged to Wellcome (inc VAT when charged)']
print ('Journal of Biological Chemistry stats')
print (Journal_BC.describe(),'\n')

c = df[df['Journal title']=='Proceedings of the National Academy of Sciences'] 
Pnas = c['COST (£) charged to Wellcome (inc VAT when charged)']
print ('Proceedings of the National Academy of Sciences stats')
print (Pnas.describe(),'\n')

d = df[df['Journal title']=='Neuro Image'] 
NI = d['COST (£) charged to Wellcome (inc VAT when charged)']
print('Neuro Image stats')
print (NI.describe(), '\n')

e = df[df['Journal title']=='Nucleic Acids Research'] 
NAR = e['COST (£) charged to Wellcome (inc VAT when charged)']
print('Nucleic Acids Research stats') 
print (NAR.describe())

Plos One stats
count       207.000000
mean      40472.287246
std      193313.084568
min         122.310000
25%         852.480000
50%         899.340000
75%        1044.640000
max      999999.000000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64 

Journal of Biological Chemistry stats
count        64.000000
mean      32666.997031
std      175112.033900
min         265.670000
25%        1158.247500
50%        1332.660000
75%        1716.270000
max      999999.000000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64 

Proceedings of the National Academy of Sciences stats
count        37.000000
mean      27846.971622
std      164260.940970
min         206.320000
25%         625.000000
50%         734.000000
75%         809.620000
max      999999.000000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64 

Neuro Image stats
count      31.000000
mean     2212.181290
std       273.193244
min      1747.160000
25%      20